tests/components/application-connector/internal/testkit/test-api/oauth.go

```diff
@@ -0,0 +1,138 @@
+package test_api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+)
+
+type OAuthCredentials struct {
+	ClientID     string
+	ClientSecret string
+}
+
+const (
+	clientIDKey     = "client_id"
+	clientSecretKey = "client_secret"
+	grantTypeKey    = "grant_type"
+)
+
+type OauthResponse struct {
+	AccessToken string `json:"access_token"`
+	TokenType   string `json:"token_type"`
+	ExpiresIn   int    `json:"expires_in"`
+	Scope       string `json:"scope"`
+}
+
+type OAuthHandler struct {
+	clientID     string
+	clientSecret string
+	mutex        sync.RWMutex
+	tokens       map[string]bool
+}
+
+func NewOAuth(clientID, clientSecret string) OAuthHandler {
+	return OAuthHandler{
+		clientID:     clientID,
+		clientSecret: clientSecret,
+		mutex:        sync.RWMutex{},
+		tokens:       make(map[string]bool),
+	}
+}
+
+func (oh *OAuthHandler) Token(w http.ResponseWriter, r *http.Request) {
+	if ok, status, message := oh.isRequestValid(r); !ok {
+		handleError(w, status, message)
+		return
+	}
+
+	token := uuid.New().String()
+
+	oh.mutex.Lock()
+	oh.tokens[token] = true
+	oh.mutex.Unlock()
+
+	response := OauthResponse{AccessToken: token, TokenType: "bearer", ExpiresIn: 3600, Scope: "basic"}
+
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(response); err != nil {
+		handleError(w, http.StatusInternalServerError, "Failed to encode token response")
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (oh *OAuthHandler) BadToken(w http.ResponseWriter, r *http.Request) {
+	if ok, status, message := oh.isRequestValid(r); !ok {
+		handleError(w, status, message)
+		return
+	}
+
+	token := uuid.New().String()
+
+	response := OauthResponse{AccessToken: token, TokenType: "bearer", ExpiresIn: 3600, Scope: "basic"}
+
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(response); err != nil {
+		handleError(w, http.StatusInternalServerError, "Failed to encode token response")
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (oh *OAuthHandler) Middleware() mux.MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			authHeader := r.Header.Get("Authorization")
+			if authHeader == "" {
+				handleError(w, http.StatusForbidden, "Authorization header missing")
+				return
+			}
+
+			splitToken := strings.Split(authHeader, "Bearer")
+			if len(splitToken) != 2 {
+				handleError(w, http.StatusForbidden, "Bearer token missing")
+				return
+			}
+
+			token := strings.TrimSpace(splitToken[1])
+
+			oh.mutex.RLock()
+			_, found := oh.tokens[token]
```

Comment:
If we ignore values in the map, maybe we could have `interface{}` and `nil` inside, to signify the lack of meaning of the stored value?